buster/busterbot.py

```diff
@@ -12,27 +13,33 @@
 from buster.formatters.prompts import prompt_formatter_factory
 from buster.retriever import Retriever
 from buster.tokenizers import tokenizer_factory
+from buster.validators.base import validator_factory
 
 logger = logging.getLogger(__name__)
 logging.basicConfig(level=logging.INFO)
 
 
 @dataclass(slots=True)
 class Response:
-    completion: Completion
-    is_relevant: bool
```

Comment:
Is there any way we can have this back?  We rely on it a lot for testing and feedback

Reply:
the problem with the `is_relevant` field is that it can only be computed once the completion text has been rendered, which only happens once the generator has been called, and via the new `Validator` class:

    is_relevant = buster.validator.check_sources_used(response)

What we could perhaps do is something like:

    response = buster.process_input(question)
    is_relevant = buster.validator.check_sources_used(response)
    response.is_relevant =  buster.validator.check_sources_used(response)

What do you think?

Reply:
How about having `buster.validator.check_sources_used(response)` modify `response.is_relevant?` And `buster.validator.check_sources_used(response)` called automatically at the end of `process_input`?

Reply:
that's the thing, is at the end of `process_input`, you DO NOT want to have to render the text from the generator, because this is what causes the delay. You only want to call `check_sources_used` after the frontend has called the generator

Reply:
changed completely how we do this, buster now handles this